substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/SubstrateTargetDescription.java

```diff
@@ -50,6 +51,16 @@ public SubstrateTargetDescription(Architecture arch, boolean isMP, int stackAlig
         return EnumSet.of(AMD64.Flag.UseCountLeadingZerosInstruction, AMD64.Flag.UseCountTrailingZerosInstruction);
     }
 
+    /**
+     * We include all flags that enable AArch64 CPU instructions as we want best possible
+     * performance for the code.
+     *
+     * @return All the flags that enable AArch64 CPU instructions.
+     */
+    public static EnumSet<AArch64.Flag> allAArch64Flags() {
+        return EnumSet.of(AArch64.Flag.UseBarriersForVolatile);
+    }
```

Comment:
I think this will always enable "UseBarriersForVolatile" . So for SVM, I think we could deal with this flag in two ways:
1. Always disable "UseBarriersForVolatile". That means always using "ldar/stlr" for volatile access.  Just never add `AArch64.Flag.UseBarriersForVolatile` to the EnumSet here.  This could avoid the issues caused by mixing use memory barriers like issues mentioned in https://github.com/oracle/graal/issues/2045.

2. Add an option for SVM like hotspot `-XX:+UseBarriersForVolatile`. And add this flag accord to the option. I also added the same option for SVM in my PR: https://github.com/oracle/graal/pull/2181.

Reply:
Right. I got that backward. Thanks for catching it. I will fix it.

So does it sound ok if I go with 1. as you suggest and let you take care of 2. with PR #2181?

Reply:
I think it's better to always use "ldar/stlr' for volatile access in SVM. So any opinion about it? @adinn @christianwimmer  Thanks!

Reply:
Yes, that's probably a better default. This was always really meant to be a diagnostic flag. However, we made it a product flag to provide an escape hatch just in case we ran into issues with early HW releases where ldar/stlr did not behave as expected. Since SVM does not have to worry about what setting is picked by an underlying JVM and since we know that all important current HW impementations of ldar/stlr are reliable it makes sense to always make this false.

Reply:
I reverted that part of the patch